Update rancher/rancher version

STARRY-S opened the pull request:
Related issue:
https://github.com/cnrancher/pandaria/issues/2860

----

Driver 的 [这里](https://github.com/cnrancher/kontainer-engine-driver-huawei/blob/master/driver/driver.go#L266 ) 使用的 `GenerateServiceAccountToken` 方法调用了 Rancher 的 [EnsureSecretForServiceAccount](https://github.com/rancher/rancher/blob/release/v2.7/pkg/kontainer-engine/drivers/util/utils.go#L98) 生成 `secret`。
本仓库之前使用的 Rancher 依赖的版本比较低，没有对 K8S 1.25 集群进行适配，因此升级依赖中的 Rancher 版本。

STARRY-S commented:
自测创建 1.25 集群成功：
<img width="1483" alt="Screenshot 2023-03-22 at 17 04 05" src="https://user-images.githubusercontent.com/41445339/226853247-9445a26b-98c3-4b23-b4e7-1b862ed6b19c.png">
<img width="1173" alt="Screenshot 2023-03-22 at 17 04 16" src="https://user-images.githubusercontent.com/41445339/226853255-33da41f2-e3eb-4643-805e-a2bf09e535cc.png">

STARRY-S commented:
@Jason-ZW @smallteeths 麻烦帮忙 Review 此 PR，谢谢！

----

改动为：
将 `github.com/rancher/rancher` 的版本更新至 `v0.0.0-20220308000653-f2a4147cf98d` 之后执行 `go mod tidy` 和 `go mod vendor`。

smallteeths reviewed (approved):
LGTM

Jason-ZW commented on go.mod:
```diff
@@ -3,8 +3,8 @@ module github.com/rancher/kontainer-engine-driver-huawei
 replace (
-	github.com/rancher/rancher/pkg/apis => github.com/rancher/rancher/pkg/apis v0.0.0-20220308000653-f2a4147cf98d
-	github.com/rancher/rancher/pkg/client => github.com/rancher/rancher/pkg/client v0.0.0-20220308000653-f2a4147cf98d
+	github.com/rancher/rancher/pkg/apis => github.com/rancher/rancher/pkg/apis v0.0.0-20230124173332-bb1c35fc4258
+	github.com/rancher/rancher/pkg/client => github.com/rancher/rancher/pkg/client v0.0.0-20230124173332-bb1c35fc4258
```
这里为什么要对标到2.7.1？而且还是Upstream版本。

STARRY-S replied:
## 背景

Driver 的 [这里](https://github.com/cnrancher/kontainer-engine-driver-huawei/blob/master/driver/driver.go#L266 ) 使用的 `GenerateServiceAccountToken` 方法调用了 Rancher 的 [EnsureSecretForServiceAccount](https://github.com/rancher/rancher/blob/release/v2.7/pkg/kontainer-engine/drivers/util/utils.go#L98) 生成 `secret`。
本仓库之前使用的 Rancher 依赖的版本比较低，没有对 K8S 1.25 集群进行适配，因此升级依赖中的 Rancher 版本。

## 改动

Rancher 最新的 stable release 为 v2.7.1，因 Rancher 仓库不符合 go 的 Major version 规范，参考[这里](https://golang.cafe/blog/how-to-upgrade-to-a-major-version-in-go.html)，需要将依赖的版本指定为 v2.7.1 所对应的 Git Commit，因此 go.mod 中填写的为:

```
github.com/rancher/rancher v0.0.0-20230124173332-bb1c35fc4258
```

这里使用 Upstream 版本是因为 RPM GC 仓库是私有仓库，确保本仓库所使用的依赖所调用的 API 是最新的就可以，因此为 v2.7.1 版本。

STARRY-S replied:
如有不足，麻烦 @smallteeths 帮忙补充。

Jason-ZW commented on go.mod:
```diff
@@ -36,21 +36,21 @@ replace (
-	github.com/sirupsen/logrus v1.8.1
-	k8s.io/api v0.23.3
-	k8s.io/apimachinery v0.23.3
+	github.com/rancher/rancher v0.0.0-20230124173332-bb1c35fc4258
```
同样问题

Jason-ZW reviewed (changes requested):
这个项目是否考虑优化下包括CI？不将vendor目录上传？

STARRY-S commented:
> 这个项目是否考虑优化下包括CI？不将vendor目录上传？

已移除了 `vendor` 文件夹。

STARRY-S commented:
已对 CI 部分做了改动，移除了 vendor 目录，rancher/rancher 依赖 replace 为 `github.com/cnrancher/pandaria`。 
使用 SUSE SLE BCI golang 镜像编译。
PR 的 CI 没有通过的原因是 github_token secret 环境变量是无法在 PR 中获取的，自测是能跑通 `make ci` 的流程。

STARRY-S committed:
Replace rancher/rancher with cnrancher/pandaria

STARRY-S committed:
Remove go mod vendor files

STARRY-S committed:
Update Drone CI

- Add OAUTH_TOKEN env to get private dependency
- Update to use SUSE BCI image
- Bump dapper to v0.6.0

Jason-ZW reviewed (approved):
LGTM
